HLTrigger/Timer/src/FastTimerService.cc

```diff
@@ -38,1425 +39,1365 @@
-    for (auto const & name: tns.getTrigPaths()) {
-      out << "FastReport              "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_active        / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_premodules    / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_intermodules  / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_postmodules   / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_overhead      / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_total         / (double) summary.count << "  "
-          << name << '\n';
-    }
-    out << '\n';
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active       Pre-     Inter-  Post-mods   Overhead      Total  EndPath" << '\n';
-    for (auto const & name: tns.getEndPaths()) {
-      out << "FastReport              "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_active        / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_premodules    / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_intermodules  / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_postmodules   / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_overhead      / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_total         / (double) summary.count << "  "
-          << name << '\n';
-    }
-  }
-  out << '\n';
-  if (m_enable_timing_modules) {
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active  Module" << '\n';
-    for (auto & keyval: m_stream.modules) {
-      std::string const & label  = keyval.first;
-      ModuleInfo  const & module = keyval.second;
-      out << "FastReport              " << std::right << std::setw(10) << module.summary_active  / (double) summary.count << "  " << label << '\n';
-    }
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active  Module" << '\n';
-    out << '\n';
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active  Module" << '\n';
-    for (auto & keyval: m_stream.moduletypes) {
-      std::string const & label  = keyval.first;
-      ModuleInfo  const & module = keyval.second;
-      out << "FastReport              " << std::right << std::setw(10) << module.summary_active  / (dou
```

Comment:
OK, it does look like threads are being swapped in and out.
I've changed the code to use `tbb::enumerable_thread_specific<Measurement>`, so it can support an arbitrary number of threads, and reproduced the logs.

There are the line ranges of the log file where each thread is used:
```
0th thread 140223481818240 used in range    513 - 207612
1st thread 140221795268352 used in range    525 - 207588
2nd thread 140221806274304 used in range    528 -  13908
3rd thread 140221784778496 used in range    533 - 207540
4th thread 140221773510400 used in range    536 - 207604
5th thread 140221752530688 used in range    551 -  13926
6th thread 140221816764160 used in range    565 - 207558
7th thread 140221763020544 used in range    649 - 207594
8th thread 140216454346496 used in range  13927 - 207568
9th thread 140216443856640 used in range  14063 - 207532

```

It looks like the 2nd and 5th threads were dropped at some point, and replaced by new threads.